index.bs

```diff
@@ -1270,6 +1270,38 @@ partial interface MLGraphBuilder {
     that all the inputs concatenated along. The size of that dimension is
     computed as the sum of all the input sizes of the same dimension.
 </div>
+<div algorithm=concat>
+    The {{MLGraphBuilder/concat(inputs, axis)}} steps are:
+    <div class="note">
+        The permissions and context validity have been checked by [[#api-mlgraphbuilder-constructor]] steps.
+    </div>
+    1. Let |inputs| be the first argument.
+        1. [=Assert=]: the type of |inputs| is sequence of {{MLOperand}} objects.
+        1. [=Assert=]: the type of |axis| is `unsigned long`.
+        1. [=Assert=]: the shape, i.e. {{MLOperandDescriptor/dimensions}}) of each operand in |inputs| is the same, except on the dimension given by |axis| on which they are concatenated.
+        1. [=Assert=]: the {{MLOperandDescriptor/type}} of each operand in |inputs| is the same.
+        1. If any of the following steps fail, then throw a "{{DataError}}" {{DOMException}} and stop.
+            1. If |inputs| is not an array of [=objects=], fail.
+            1. If |axis| is not a positive integer [=number=], fail.
+            1. If |axis| is greater than or equal to the <a>rank</a> of |inputs|, fail.
+            1. Let |desc| be |inputs|[0].{{MLOperand/[[descriptor]]}}.
+            1. Let |desc|.{{MLOperandDescriptor/dimensions}}[|axis|] be `0`.
+            1. For each |index| between 0 and the <a>rank</a> of |inputs|:
+                1. If running <a>validate MLOperand</a> given |inputs|[|index|] and [=this=]  returns `false`, then fail.
+                1. For each |dim| between 0 and the <a>rank</a> of |inputs|[|index|]:
+                    <div class="note">
+                        If the shape (each corresponding dimension and type) of the operands in |inputs| is not the same, except on the dimension given by |axis| (on which they are concatenated), fail.
```

Comment:
Consider a reword: If the shape of each corresponding dimension and type of the operands, except for those of the dimension given by |axis|, is not the same, fail.